internal/kafka/internal/services/clusters.go

```diff
@@ -108,6 +110,21 @@ func (c clusterService) RegisterClusterJob(clusterRequest *api.Cluster) *apiErro
 	return nil
 }
 
+// DeregisterClusterJob deregister an existing cluster
+func (c clusterService) DeregisterClusterJob(clusterId string) *apiErrors.ServiceError {
+	cluster, err := c.FindClusterByID(clusterId)
+	if err != nil {
+		return err
+	}
+
+	updateErr := c.UpdateStatus(*cluster, api.ClusterDeprovisioning)
+	if updateErr != nil {
+		return apiErrors.NewWithCause(apiErrors.ErrorGeneral, err, "failed to deregister cluster")
+	}
+
+	return nil
```

Comment:
```suggestion
	return updateErr
```

Reply:
I prefer the previous approach, so we are sure we return a naked nil.
That will help when/if we will start returning `error` instead of this pointer

Reply:
Agree with @ziccardi 
Returning `updateErr` makes it look like an error is returned even though its assumed to be nil here.